### PR TITLE
Automate 1531, 1296 Pass filters

### PR DESCRIPTION
**Pull request by tarablack01**
### :nut_and_bolt: Description: What code changed, and why?
1531
When filters are applied through the search bar,
update the count in the health status filters above the table
update the pagination under the table
1296
When using the filter bar, we want to only show matching services on the right side detail panel. This should help users focus on only the services they are actually interested in; for example, when filtering on site, users only wish to see services from the given site.

### :chains: Related Resources
https://app.zenhub.com/workspaces/ui-team-5cba23a3b14fdc05970c8f87/issues/chef/automate/1296
https://app.zenhub.com/workspaces/ui-team-5cba23a3b14fdc05970c8f87/issues/chef/automate/1531

### :+1: Definition of Done
1296
A service group where all services match the given criteria is shown unchanged compared to the unfiltered view, and all of its services appear in the right-side panel;
A service group where some of the services match is shown in the left side table, but the content is modified to include only the services matching the filter criteria; that is the Health, Service Group and Release columns reflect only the services that match the filter criteria. The right side services list should show only the services that matched the filter criteria.

### :athletic_shoe: How to Build and Test the Change
hab studio enter
build components/automate-ui-devproxy
start_automate_ui_background
start_all_services
applications_populate_database

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [x] Docs added/updated?

### :camera: Screenshots, if applicable
Screen Shot 2019-09-10 at 9 00 44 PM
Screen Shot 2019-09-10 at 9 02 28 PM

### CODE REVIEW

File: components/automate-ui/src/app/pages/service-groups/service-groups.component.ts [hunk 1]
```diff
@@ -294,11 +294,20 @@ export class ServiceGroupsComponent implements OnInit, OnDestroy {
   detailParamsChange(queryParams) {
     const sgId = queryParams.get('sgId');
     if (sgId) {
+      const searchBarFilters = [];
+      const paramKeys = Object.keys(queryParams.params);
+      paramKeys.forEach((key: string) => {
+        if (some({'type': key}, this.categoryTypes)) {
+          searchBarFilters.push({ type: key, text: queryParams.params[key] });
```
Comment (srenatus) [inline components/automate-ui/src/app/pages/service-groups/service-groups.component.ts]: I'd prefer 
```typescript
const searchBarFilters = paramKeys.filter((key: string) => some({'type': key}, this.categoryTypes))
  .map((key: string) => ({ type: key, text: queryParams.params[key] }));
```
but it's just a matter of taste 😃

File: components/automate-ui/src/app/entities/service-groups/service-groups.effects.ts [hunk 2]
```diff
@@ -64,9 +64,9 @@ export class ServiceGroupsEffects {
   @Effect()
   getServiceGroupsCounts$ = this.actions$.pipe(
       ofType(ServiceGroupsActionTypes.GET_SERVICE_GROUPS_COUNTS),
-      withLatestFrom(this.store),
-      switchMap(([_action]) => {
-        return this.requests.fetchServiceGroupHealth().pipe(
+      withLatestFrom(this.store.select(serviceGroupsFilters)),
+      switchMap(([_action, filters]: [any, ServiceGroupsFilters]) => {
+        return this.requests.fetchServiceGroupHealth(filters).pipe(
         map((payload: ServiceGroupsHealthSummary) => new GetServiceGroupsCountsSuccess(payload)),
```
Comment (srenatus) [inline components/automate-ui/src/app/entities/service-groups/service-groups.effects.ts]: [nit] Is indentation correct here? I'd expect a level of nesting more. 🤔

File: components/automate-ui/src/app/pages/service-groups/service-groups.component.ts (hunk 1 above)
Comment (tarablack01) [reply components/automate-ui/src/app/pages/service-groups/service-groups.component.ts]: @srenatus I like it 👍🏻

File: components/automate-ui/src/app/entities/service-groups/service-groups.effects.ts (hunk 2 above)
Comment (tarablack01) [reply components/automate-ui/src/app/entities/service-groups/service-groups.effects.ts]: @srenatus Updated the indentation :)

**Comment by apriofrost** [comment]
A side note: the wildcarding behavior also worked as intended. 👍

**Pull request merged**